website/server/libs/oauth.js

```diff
@@ -0,0 +1,103 @@
+import { 
+    OAuthToken as OAuthTokensModel, 
+    OAuthClients as OAuthClientsModel,
+    OAuthCode as OAuthCodeModel
+} from '../models/oauth';
+import {
+  model as User,
+} from '../models/user';
+import validator from 'validator';
+import _ from 'lodash';
+import moment from 'moment';
+import { v4 as uuid } from 'uuid';
+import { createServer, grant, exchange } from 'oauth2orize';
+import { removeFromArray } from './collectionManipulators';
+
+// Create OAuth 2.0 server
+const server = createServer();
+
+// Register serialialization and deserialization functions.
+//
+// When a client redirects a user to user authorization endpoint, an
+// authorization transaction is initiated. To complete the transaction, the
+// user must authenticate and approve the authorization request. Because this
+// may involve multiple HTTP request/response exchanges, the transaction is
+// stored in the session.
+//
+// An application must supply serialization functions, which determine how the
+// client object is serialized into the session. Typically this will be a
+// simple matter of serializing the client's ID, and deserializing by finding
+// the client by ID from the database.
+
+server.serializeClient((client, done) => done(null, client.clientId));
+
+server.deserializeClient(async (clientId, done) => {
```

Comment:
mixing async/await and callbacks usually causes problems, would you mind turning the `await User...` to a standard promise? `User.find(...).exec().then(...)`?